### ISSUE TITLE
PcapLiveDevicebyIP (or name) return null ptr

### ISSUE THREAD

**Issue by quynhngatt** [seladb/PcapPlusPlus#961]
Hello,
I want to send packets to another device.
When I try getPcapLiveDeviceByIp by my laptop IP, it's work ok. But it return null ptr when i get by another IP that connected the same network. Please tell me why T.T

**Comment by seladb** [comment]
`getPcapLiveDeviceByIp()` returns a device which is related to a NIC. For example, if the NIC on your machine has IP address of `10.0.0.1` then `getPcapLiveDeviceByIp("10.0.0.1")` will return this device. Now you can use this device (NIC) to send or receive packets.
Please let me know it this helps.

**Comment by quynhngatt** [comment]
Thanks for your quick reply.
I got it but i have no idea to send packet to another device :<

**Comment by seladb** [comment]
Sending packets between devices is a broad topic and for the most part can be achieved by tools which are not PcapPlusPlus.
First, it depends on what kind of communication you're interested in. For example: if it's TCP/UDP you can probably open a socket, if it's HTTP/S you probably need a web-server and a web-client, for ICMP you can use the `ping` command, etc.

PcapPlusPlus mostly helps with parsing of incoming packets, or creating packets from scratch and sending them to another machine. You can look at examples like [DNSResolver](https://github.com/seladb/PcapPlusPlus/tree/master/Examples/DNSResolver) which creates and sends DNS requests - a simple version of the `host` command, or [Arping](https://github.com/seladb/PcapPlusPlus/tree/master/Examples/Arping) that does the same for ARP requests, which is quite similar to the Linux `arping` command.

To summarize - it really depends on your use-case.

**Comment by quynhngatt** [comment]
thankyou very much.

**Comment by seladb** [comment]
I'll close this issue now, please reopen if needed
